gm4_vecto_shamir/data/gm4_vecto_shamir/predicates/sneak_validblock.json

```diff
@@ -0,0 +1,63 @@
+[
+  {
+    "condition": "minecraft:entity_properties",
+    "entity": "this",
+    "predicate": {
+      "flags": {
+        "is_sneaking": true
+      },
+      "equipment": {
+        "feet": {
+          "tag": "gm4:boots",
+          "nbt": "{gm4_metallurgy:{has_shamir:1b,active_shamir:'vecto'}}"
+        }
+      }
+    }
+  },
+  {
+    "condition": "minecraft:alternative",
+    "terms": [
+      {
+        "condition": "minecraft:location_check",
+        "predicate": {
+          "block": {
+            "tag": "minecraft:rails"
+          }
+        }
+      },
+      {
+        "condition": "minecraft:location_check",
+        "offsetX": 0,
+        "offsetY": -1,
+        "offsetZ": 0,
+        "predicate": {
+          "block": {
+            "tag": "minecraft:ice"
+          }
+        }
+      },
+      {
+        "condition": "minecraft:location_check",
+        "offsetX": 0,
+        "offsetY": -1,
+        "offsetZ": 0,
+        "predicate": {
+          "fluid": {
+            "tag": "minecraft:water"
+          }
+        }
+      }
+    ]
+  },
+  {
+    "condition": "minecraft:inverted",
+    "term": {
+      "condition": "minecraft:location_check",
+      "predicate": {
+        "fluid": {
+          "tag": "minecraft:water"
```

Comment:
These should probably use `#gm4:water` instead as there are some water-like blocks that aren't actually water, e.g. bubble columns or seagrass.

Reply:
> These should probably use `#gm4:water` instead as there are some water-like blocks that aren't actually water, e.g. bubble columns or seagrass.

Both are checks for "water fluid", so any water-like block should be detected. I assume it's the same as both #gm4:water and #gm4:waterloggable combined. The first check is to make sure there's water below the player and the second check is to make sure you're not underwater.
The reason it checks for any "water fluid" here, is that this predicate is checked from main, so I assumed having more block checks.

Reply:
That makes sense! Sorry, I don't know enough about minecraft's barebones fluid system.